src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooBuildPlanService.java

```diff
@@ -183,6 +183,26 @@ private Stage createBuildStage(ProgrammingLanguage programmingLanguage, final bo
                 tasks.add(0, checkoutTask);
                 return defaultStage.jobs(defaultJob.tasks(tasks.toArray(new Task[0])).finalTasks(testParserTask));
             }
+            case VHDL -> {
+                // Do not run the builds in extra docker containers if the dev-profile is active
+                if (!activeProfiles.contains(JHipsterConstants.SPRING_PROFILE_DEVELOPMENT)) {
+                    defaultJob.dockerConfiguration(new DockerConfiguration().image("tizianleonhardt/era-artemis-vhdl:latest"));
+                }
+                final var testParserTask = new TestParserTask(TestParserTaskProperties.TestType.JUNIT).resultDirectories("**/result.xml");
+                var tasks = readScriptTasksFromTemplate(programmingLanguage, sequentialBuildRuns);
+                tasks.add(0, checkoutTask);
+                return defaultStage.jobs(defaultJob.tasks(tasks.toArray(new Task[0])).finalTasks(testParserTask));
+            }
+            case ASSEMBLER -> {
+                // Do not run the builds in extra docker containers if the dev-profile is active
+                if (!activeProfiles.contains(JHipsterConstants.SPRING_PROFILE_DEVELOPMENT)) {
+                    defaultJob.dockerConfiguration(new DockerConfiguration().image("tizianleonhardt/era-artemis-assembler:latest"));
+                }
+                final var testParserTask = new TestParserTask(TestParserTaskProperties.TestType.JUNIT).resultDirectories("**/result.xml");
+                var tasks = readScriptTasksFromTemplate(programmingLanguage, sequentialBuildRuns);
+                tasks.add(0, checkoutTask);
+                return defaultStage.jobs(defaultJob.tasks(tasks.toArray(new Task[0])).finalTasks(testParserTask));
+            }
             default -> throw new IllegalArgumentException("No build stage setup for programming language " + programmingLanguage);
```

Comment:
```suggestion
```
We should delete this to let the compiler check the correctness, I think this would be better.

Reply:
when we cover all cases before, then yes, we should delete the default statement

Reply:
I've removed the default branch.

Reply:
@krusche @MaisiKoleni The removal of the default branch caused an error (I believe something with switch expressions vs statements). As I will refactor the code in a follow-up, I've reverted it for now (see #2150).